storm-server/src/main/java/org/apache/storm/stats/StatsUtil.java

```diff
@@ -2004,7 +2005,7 @@ private static ComponentPageInfo thriftifyCompPageData(
         win2stats.put(FAILED, ClientStatsUtil.getMapByKey(data, WIN_TO_FAILED));
 
         String compType = (String) data.get(TYPE);
-        if (compType.equals(ClientStatsUtil.SPOUT)) {
+        if (Objects.equals(compType,ClientStatsUtil.SPOUT)) {
```

Comment:
It looks like only avoiding NPE, which also makes misleading as component being interpreted as bolt. We should see what are the actual cases of `data.get(TYPE)` being null, and decide how to fix based on the case.

Reply:
I agree. Maybe also consider throwing exceptions if it will ever be null. Need to understand it more